components/match2/wikis/counterstrike/match_summary.lua

```diff
@@ -0,0 +1,499 @@
+		end
+
+		return '[[FILE:' .. _icon .. '|link=' .. url .. '|15px|' .. label .. '|alt=' .. url .. ']]'
+	end
+
+	-- Match vod
+	if not Logic.isEmpty(match.vod) then
+		footer:addElement(VodLink.display{
+			vod = match.vod,
+			source = match.vod.url
+		})
+	end
+
+	-- Game Vods
+	for index, vod in pairs(vods) do
+		-- TODO: Darkmode VodIcons
+		footer:addElement(VodLink.display{
+			gamenum = index,
+			vod = vod,
+			source = vod.url
+		})
+	end
+
+	if Table.isNotEmpty(match.links) then
+		if Logic.isNotEmpty(vods) or match.vod then
+			footer:addElement(separator)
+		end
+	else
+		return footer
+	end
+
+	--- Platforms is used to keep the order of the links in footer
+	local platforms = mw.loadData('Module:MatchExternalLinks')
+	local links = match.links
+
+	local insertDotNext = false
+	local iconsInserted = 0
+
+	for _, platform in ipairs(platforms) do
+		if Logic.isNotEmpty(platform) then
+			local link = links[platform.name]
+			if link then
+				if insertDotNext then
+					insertDotNext = false
+					footer:addElement(separator)
+				end
+
+				local icon = platform.icon
+				local label = platform.label
+				local addGameLabel = platform.isMapStats and match.bestof and match.bestof > 1
+
+				for _, val in ipairs(link) do
+					footer:addElement(createFooterLink(icon, val[1], label,
+														addGameLabel and val[2] or 0))
+					iconsInserted = iconsInserted + 1
+				end
+
+				if platform.stats then
+					for _, site in ipairs(platform.stats) do
+						if links[site] then
+							footer:addElement(separator)
+							break
+						end
+					end
+				end
+			end
+		else
+			insertDotNext = iconsInserted > 0 and true or false
+			iconsInserted = 0
+		end
+	end
+
+	return footer
+end
+
+function CustomMatchSummary._createMap(game)
+	local row = MatchSummary.Row()
+	local extradata = game.extradata or {}
+
+	-- Score
+	local team1Score = Score():setLeft()
+	local team2Score = Score():setRight()
+
+	-- Score Team 1
+	team1Score:setMapScore(game.scores[1])
+
+	local t1sides = extradata['t1sides'] or {}
+	local t2sides = extradata['t2sides'] or {}
+	local t1halfs = extradata['t1halfs'] or {}
+	local t2halfs = extradata['t2halfs'] or {}
+
+	local numberOfSides = #t2sides
+
+	-- Insert team scores
+	for sideIndex, side in ipairs(t1sides) do
+		-- Team 1 scores inserted from 1 .. n
+		if math.fmod(sideIndex, 2) == 1 then
+			team1Score:setFirstHalfScore(t1halfs[sideIndex], CustomMatchSummary._getSideColor(side))
+		else
+			team1Score:setSecondHalfScore(t1halfs[sideIndex], CustomMatchSummary._getSideColor(side))
+		end
+
+		-- Team 2 scores inserted from n .. 1
+		local t2SideIndex = numberOfSides - sideIndex + 1
+		local oppositeSide = t2sides[t2SideIndex]
+		if math.fmod(t2SideIndex, 2) == 1 then
+			team2Score:setFirstHalfScore(t2halfs[t2SideIndex], CustomMatchSummary._getSideColor(oppositeSide))
+		else
+			team2Score:setSecondHalfScore(t2halfs[t2SideIndex], CustomMatchSummary._getSideColor(oppositeSide))
```

Comment:
side is just `'t'`/`'ct'`, right?
if you do the change i proposed above (or something similar)
then you can just pass the side here

```suggestion
			team1Score:setFirstHalfScore(t1halfs[sideIndex], side)
		else
			team1Score:setSecondHalfScore(t1halfs[sideIndex], side)
		end

		-- Team 2 scores inserted from n .. 1
		local t2SideIndex = numberOfSides - sideIndex + 1
		local oppositeSide = t2sides[t2SideIndex]
		if math.fmod(t2SideIndex, 2) == 1 then
			team2Score:setFirstHalfScore(t2halfs[t2SideIndex], oppositeSide)
		else
			team2Score:setSecondHalfScore(t2halfs[t2SideIndex], oppositeSide)
```

Reply:
Yap sides are only 'ct' and 't'